diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c215c0..7c2ad33 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -185,5 +185,7 @@ endif()
 #
 # build the library
 #
-
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
 add_subdirectory(src)
@@ -224,9 +226,6 @@ endif()
 # install
 #
-
-include(GNUInstallDirs)
-include(CMakePackageConfigHelpers)
-
+
 set(LLAMA_INCLUDE_INSTALL_DIR ${CMAKE_INSTALL_INCLUDEDIR} CACHE PATH "Location of header  files")
 set(LLAMA_LIB_INSTALL_DIR     ${CMAKE_INSTALL_LIBDIR}     CACHE PATH "Location of library files")
 set(LLAMA_BIN_INSTALL_DIR     ${CMAKE_INSTALL_BINDIR}     CACHE PATH "Location of binary  files")
@@ -239,6 +239,6 @@ set_target_properties(llama
     PROPERTIES
         PUBLIC_HEADER "${LLAMA_PUBLIC_HEADERS}")
-
-install(TARGETS llama LIBRARY PUBLIC_HEADER)
+
+install(TARGETS llama EXPORT llamaTargets LIBRARY PUBLIC_HEADER)
-
+
 configure_package_config_file(
@@ -257,5 +257,10 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/llama-config.cmake
               ${CMAKE_CURRENT_BINARY_DIR}/llama-version.cmake
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/llama)
-
+
+install(EXPORT llamaTargets
+        FILE llama-targets.cmake
+        NAMESPACE llama::
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/llama)
+
 install(
     FILES convert_hf_to_gguf.py
diff --git a/cmake/llama-config.cmake.in b/cmake/llama-config.cmake.in
index 90cbec5..21d920a 100644
--- a/cmake/llama-config.cmake.in
+++ b/cmake/llama-config.cmake.in
@@ -6,25 +6,13 @@ set(LLAMA_SHARED_LIB   @BUILD_SHARED_LIBS@)
 @PACKAGE_INIT@
-
+
+include(CMakeFindDependencyMacro)
+
 set_and_check(LLAMA_INCLUDE_DIR "@PACKAGE_LLAMA_INCLUDE_INSTALL_DIR@")
-set_and_check(LLAMA_LIB_DIR     "@PACKAGE_LLAMA_LIB_INSTALL_DIR@")
-set_and_check(LLAMA_BIN_DIR     "@PACKAGE_LLAMA_BIN_INSTALL_DIR@")
-
-find_package(ggml REQUIRED HINTS ${LLAMA_LIB_DIR}/cmake)
-
-find_library(llama_LIBRARY llama
-    REQUIRED
-    HINTS ${LLAMA_LIB_DIR}
-    NO_CMAKE_FIND_ROOT_PATH
-)
-
-add_library(llama UNKNOWN IMPORTED)
-set_target_properties(llama
-    PROPERTIES
-        INTERFACE_INCLUDE_DIRECTORIES "${LLAMA_INCLUDE_DIR}"
-        INTERFACE_LINK_LIBRARIES "ggml::ggml;ggml::ggml-base;"
-        IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
-        IMPORTED_LOCATION "${llama_LIBRARY}"
-        INTERFACE_COMPILE_FEATURES c_std_90
-        POSITION_INDEPENDENT_CODE ON)
-
+
+find_dependency(ggml CONFIG)
+find_dependency(Threads)
+find_dependency(miniaudio CONFIG)
+
+include("${CMAKE_CURRENT_LIST_DIR}/llama-targets.cmake")
+
 check_required_components(Llama)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 54bb6ce..7a9af16 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -140,10 +140,13 @@ set_target_properties(llama PROPERTIES
 )
-
+
 target_include_directories(llama PRIVATE .)
-target_include_directories(llama PUBLIC ../include)
+target_include_directories(llama PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>"
+    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+)
 target_compile_features   (llama PRIVATE cxx_std_17) # don't bump
-
-target_link_libraries(llama PUBLIC ggml)
-
+
+target_link_libraries(llama PUBLIC ggml::ggml)
+
 if (BUILD_SHARED_LIBS)
     set_target_properties(llama PROPERTIES POSITION_INDEPENDENT_CODE ON)
diff --git a/tools/mtmd/CMakeLists.txt b/tools/mtmd/CMakeLists.txt
index 2a10339..a51a02d 100644
--- a/tools/mtmd/CMakeLists.txt
+++ b/tools/mtmd/CMakeLists.txt
@@ -1,7 +1,8 @@
 # mtmd
-
+
 find_package(Threads REQUIRED)
+find_package(miniaudio CONFIG REQUIRED)
-
+
 add_library(mtmd
             mtmd.cpp
             mtmd-audio.cpp
@@ -14,11 +15,14 @@ add_library(mtmd
 set_target_properties(mtmd PROPERTIES
     VERSION ${LLAMA_INSTALL_VERSION}
     SOVERSION 0
 )
-
-target_link_libraries     (mtmd PUBLIC ggml llama)
-target_link_libraries     (mtmd PRIVATE Threads::Threads)
-target_include_directories(mtmd PUBLIC  .)
+
+target_link_libraries     (mtmd PUBLIC ggml::ggml llama)
+target_link_libraries     (mtmd PRIVATE Threads::Threads miniaudio::miniaudio)
+target_include_directories(mtmd PUBLIC
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
+    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+)
 target_include_directories(mtmd PRIVATE ../..)
 target_include_directories(mtmd PRIVATE ../../vendor)
 target_compile_features   (mtmd PRIVATE cxx_std_17)
@@ -39,9 +43,9 @@ set_target_properties(mtmd
     PROPERTIES
     PUBLIC_HEADER "${MTMD_PUBLIC_HEADERS}")
-
-install(TARGETS mtmd LIBRARY PUBLIC_HEADER)
-
+
+install(TARGETS mtmd EXPORT llamaTargets LIBRARY PUBLIC_HEADER)
+
 if (NOT MSVC)
-    # for stb_image.h and miniaudio.h
+    # for stb_image.h
     target_compile_options(mtmd PRIVATE -Wno-cast-qual)
 endif()
diff --git a/tools/mtmd/mtmd-helper.cpp b/tools/mtmd/mtmd-helper.cpp
index 6e977a4..597996b 100644
--- a/tools/mtmd/mtmd-helper.cpp
+++ b/tools/mtmd/mtmd-helper.cpp
@@ -16,18 +16,6 @@
 #include <vector>
-
-//#define MTMD_AUDIO_DEBUG
-
-#define MINIAUDIO_IMPLEMENTATION
-#ifndef MTMD_AUDIO_DEBUG
-#   define MA_NO_ENCODING
-#endif
-#define MA_NO_DEVICE_IO
-#define MA_NO_RESOURCE_MANAGER
-#define MA_NO_NODE_GRAPH
-#define MA_NO_ENGINE
-#define MA_NO_GENERATION
-#define MA_API static
-#include "miniaudio/miniaudio.h"
-
+
+#include <miniaudio.h>
+
 #define STB_IMAGE_IMPLEMENTATION
 #include "stb/stb_image.h"
